src/routes/CreateNewSafePage/components/SafeCreationProcess.tsx

```diff
@@ -0,0 +1,240 @@
+import { ReactElement, useState, useEffect, useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { backOff } from 'exponential-backoff'
+import { TransactionReceipt } from 'web3-core'
+import { generatePath } from 'react-router'
+import { GenericModal } from '@gnosis.pm/safe-react-components'
+import { makeStyles } from '@material-ui/core/'
+
+import { getSafeDeploymentTransaction } from 'src/logic/contracts/safeContracts'
+import { txMonitor } from 'src/logic/safe/transactions/txMonitor'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { FIELD_CREATION_PROXY_SALT } from 'src/routes/open/components/fields'
+import { SafeDeployment } from 'src/routes/opening'
+import { SAFE_ROUTES, WELCOME_ADDRESS } from 'src/routes/routes'
+import { history } from 'src/store'
+import { useAnalytics } from 'src/utils/googleAnalytics'
+import { loadFromStorage, removeFromStorage, saveToStorage } from 'src/utils/storage'
+import { sleep } from 'src/utils/timer'
+import { addOrUpdateSafe } from 'src/logic/safe/store/actions/addOrUpdateSafe'
+import {
+  SAFE_PENDING_CREATION_STORAGE_KEY,
+  CreateSafeFormValues,
+  FIELD_NEW_SAFE_THRESHOLD,
+  FIELD_SAFE_OWNERS_LIST,
+  FIELD_NEW_SAFE_GAS_LIMIT,
+  FIELD_NEW_SAFE_CREATION_TX_HASH,
+  FIELD_CREATE_SUGGESTED_SAFE_NAME,
+  FIELD_CREATE_CUSTOM_SAFE_NAME,
+} from '../fields/createSafeFields'
+import { getSafeInfo } from 'src/logic/safe/utils/safeInformation'
+import { buildSafe } from 'src/logic/safe/store/actions/fetchSafe'
+import { addressBookSafeLoad } from 'src/logic/addressBook/store/actions'
+import { makeAddressBookEntry } from 'src/logic/addressBook/model/addressBook'
+import Paragraph from 'src/components/layout/Paragraph'
+import NetworkLabel from 'src/components/NetworkLabel/NetworkLabel'
+import Button from 'src/components/layout/Button'
+import { boldFont } from 'src/theme/variables'
+
+type ModalDataType = {
+  safeAddress: string
+  safeName?: string
+  safeCreationTxHash?: string
+}
+
+function SafeCreationProcess(): ReactElement {
+  const [safeCreationTxHash, setSafeCreationTxHash] = useState<string | undefined>()
+  const [creationTxPromise, setCreationTxPromise] = useState<Promise<TransactionReceipt>>()
+
+  const { trackEvent } = useAnalytics()
+  const dispatch = useDispatch()
+  const userAddressAccount = useSelector(userAccountSelector)
+
+  const classes = useStyles()
+
+  const [showModal, setShowModal] = useState(false)
+  const [modalData, setModalData] = useState<ModalDataType>({ safeAddress: '' })
+
+  const createNewSafe = useCallback(async () => {
+    const safeCreationFormValues = (await loadFromStorage(SAFE_PENDING_CREATION_STORAGE_KEY)) as CreateSafeFormValues
+    setSafeCreationTxHash(safeCreationFormValues[FIELD_NEW_SAFE_CREATION_TX_HASH])
+
+    setCreationTxPromise(
+      new Promise((resolve, reject) => {
```

Comment:
Maybe move this function outside of the component? It'd be cleaner.

Reply:
Its using `userAddressAccount` and `safeCreationFormValues` so I will suggest to keep it there